tests/helpers/test_state.py

```diff
@@ -186,9 +188,10 @@ def test_call_to_component(hass):
 async def test_as_number_states(hass):
     """Test state_as_number with states."""
     zero_states = (STATE_OFF, STATE_CLOSED, STATE_UNLOCKED,
-                   STATE_BELOW_HORIZON, STATE_NOT_HOME)
+                   STATE_BELOW_HORIZON, STATE_NOT_HOME, STATE_OFFLINE,
+                   STATE_UNAVAILABLE)
     one_states = (STATE_ON, STATE_OPEN, STATE_LOCKED, STATE_ABOVE_HORIZON,
-                  STATE_HOME)
+                  STATE_HOME, STATE_ONLINE)
```

Comment:
I don't think that this is correct. Instead, all integrations that use online/offline should become binary_sensor instead of using a sensor. If there is no device class for online/offline, we should add that https://developers.home-assistant.io/docs/en/entity_binary_sensor.html#available-device-classes

Reply:
So a connectivity class for the binary sensor?

Reply:
Yeah that can work.